<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <node name="$(anon vl53l1x_scanner_driver)" pkg="vl53l1x_scanner" type="vl53l1x_scanner_driver_node"  output="screen">

        <!-- ===== [HARDWARE PARAMETERS] ===== -->
        <param name="p_scanner_serial_port" value="/dev/ttyACM0"/>
        <param name="p_stepper_horizontal_angle_per_step" value="0.003067962"/>
        <param name="p_laser_vertical_angle_per_roi_cell" value="0.035"/>
        <param name="p_laser_range_min" value="0.04"/>
        <param name="p_laser_range_max" value="4.0"/>

        <!-- ===== [ROS PARAMETERS] ===== -->
        <param name="p_laserscan_frame" value="laser"/>

        <!-- ===== [SCANNER PARAMETERS (WILL BE SENT TO THE SCANNER)] ===== -->
        <param name="p_stepper_pin_1" value="9"/>
        <param name="p_stepper_pin_2" value="10"/>
        <param name="p_stepper_pin_3" value="11"/>
        <param name="p_stepper_pin_4" value="12"/>
        <param name="p_stepper_delay" value="2250"/>
        <param name="p_stepper_step_min" value="-256"/>
        <param name="p_stepper_step_max" value="+256"/>
        <param name="p_laser_roi_topleftx" value="6"/>
        <param name="p_laser_roi_botrightx" value="9"/>
        <param name="p_laser_distance_mode" value="3"/>
        <param name="p_laser_measurement_timing_budget_micro_seconds" value="150000"/>
        <param name="p_laser_inter_measurement_period_milli_seconds" value="150"/>
        <param name="p_scanner_mode" value="3"/>
        <param name="p_scanner_vertical_steps" value="1"/>
        <param name="p_scanner_horizontal_steps" value="8"/>
        <param name="p_scanner_rewind" value="1"/>
        <param name="p_scanner_calibration_max_value" value="500"/>

    </node>
</launch>